### ISSUE TITLE
Add support for placeholders

### ISSUE THREAD

**Issue by soonick** [ebidel/i18n-msg#26]
I need translated strings with placeholders, so I looked at the i18n chrome spec(https://developer.chrome.com/extensions/i18n-messages) and created this based on it:
https://github.com/ebidel/i18n-msg/pull/25

I'm trying to follow the spec without implementing it all of it so it can probably be extended in the future.

Currently there are two ways we can use placeholders:

**Placeholders in the json file**

```json
{
  "bye": {
    "message": "Come back to $our_site$ soon!",
    "placeholders": {
      "our_site": {
        "content": "Example.com",
      }
    }
  }
}
```

These work pretty much as documented by chrome. We have a placeholders attribute that contains the placeholders. In this case there is one called *our_site* with a content of Example.com. When we find this placeholder we can do a simple string replacement. Replace $our_site$ with with the content of the *our_site* placeholder.

**Placeholders in a component attribute**

```json
{
  "bye": {
    "message": "Come back to $our_site$ soon!",
    "placeholders": {
      "our_site": {
        "content": "$1",
      }
    }
  }
}
```

In this case the content of the placeholder is $1. In this case I am currently just checking that the first character is a $. If it is then I assume you are looking for an attribute. In this example, since it is *$1* it will look for the first placeholder passed to the component($2 is the second, and so on). The placeholders are passed as an array in an attribute called placeholders:

```html
<i18n-msg msgid="seconds" placeholders='["Example.com"]'>PLACEHOLDER_STRING</i18n-msg>
```

The result in both cases is: *Come back to Example.com soon!*

**Comment by rohitkhode** [comment]
+1

**Comment by benjaminpg** [comment]
Oh my, I just sent a pull request for something similar... nonetheless I used a different aproach to insert the placeholders, to be more precise the one used in Spring MVC and its MessageSource class. Rather than label each placeholder it just use $n (n is a numeric value) as replace pattern. That way it's not necessary to declare each placeholder in your locale. On the other hand, in my case I got to parse the message to find these pattern and check if I was using it, which makes my solution less eficient. I think it would be better if I used something like "usePlaceholders: true".

**Comment by benjaminpg** [comment]
When will this feature be merged into? We are going to start a proyect with Polymer and I would like to have this before we start working.
